arcgishub/sites.py

```diff
@@ -413,11 +436,96 @@ def update(self, site_properties=None, data=None, thumbnail=None, metadata=None)
         
             >> True
         """
+        _site_data = self.definition
         if site_properties:
-            _site_data = self.definition
             for key, value in site_properties.items():
                 _site_data[key] = value
-            return self.item.update(_site_data, data, thumbnail, metadata)
+        if subdomain:
+            #format subdomain if needed
+            subdomain = subdomain.replace(' ', '-').lower()
+            #Domain manipulation for new subdomain
+            if self._gis._portal.is_arcgisonline:
+                #Check for length of domain
+                if len(subdomain + '-' + self._gis.properties['urlKey']) > 63:
+                    _num = 63 - len(self._gis.properties['urlKey'])
+                    raise ValueError('Requested url too long. Please enter a subdomain shorter than %d characters' %_num)
+                #Fetch siteId from domain entry
+                _HEADERS = {
+                    'Content-Type': 'application/json', 
+                    'Authorization': self._gis._con.token, 
+                    'Referer': self._gis._con._referer
+                }
+                path = 'https://hub.arcgis.com/utilities/domains?siteId='+self.itemid
```

Comment:
GIS stores the url for the ArcGIS organization. I'm not sure how to extract the hub api url from there.

Reply:
I updated the `Hub` object with a property that can verify if it is dev/qa/prod and return the appropriate hub API url. However, the Sites class is provided a reference to the `GIS` object instead of the `Hub` object, to support Enterprise Sites access from the Python API.
We have 3 solutions in that case:
1. Let the hub.py version of the `Sites` class work with the `Hub` object and not the `GIS` object. Downside to that is the continuous effort of maintaining two distinct versions in `arcgishub` and `arcgis`.
2.  Wait to update this method with the environment URL passed in, after the next release of `arcgis`. That will ensure that the `GIS` object can access the updated hub (with this ability to detect env) through the `gis.hub` property. Until the next release in October we will only support this for prod in hub.py.
3. Continue to support only Customer (aka prod) environments via hub.py.

Let me know which approach to proceed with.